src/main/scala/bi/deep/SchemaUtils.scala

```diff
@@ -24,13 +24,13 @@ object SchemaUtils {
   private def merge(
                      columnsA: Array[(String, ColumnCapabilitiesImpl)],
                      columnsB: Array[(String, ColumnCapabilitiesImpl)]): Array[(String, ColumnCapabilitiesImpl)] = {
-    val result = mutable.LinkedHashMap[String, ColumnCapabilitiesImpl]()
-    result ++= columnsA
+    val result = mutable.LinkedHashMap[String, ColumnCapabilitiesImpl](columnsA: _*)
 
-    columnsB.foreach { case (name, capabilities) =>
-      result += name -> ColumnCapabilitiesImpl.merge(capabilities, result.getOrElse(name, null), coercionLogic)
-    }
-    Array() ++ result
+    columnsB.foldLeft(result) { case (merged, (name, capabilities)) =>
+      merged += name -> ColumnCapabilitiesImpl.merge(capabilities, merged.getOrElse(name, null), coercionLogic)
+    }.toArray
+
+    result.toArray
```

Comment:
You ignore result returned by `foldLeft`. Fold returns an aggregated value, it aggregate starting from first element in your case `result` then it merge it with all `columnsB` values. Each loop updates the internal state that is available in merge function as `merged`. This state is also known as `accumulator` 

You can just do it like this:
```scala
   val accumulator = mutable.LinkedHashMap[String, ColumnCapabilitiesImpl](columnsA: _*)

    columnsB.foldLeft(accumulator) { case (merged, (name, capabilities)) =>
      merged += name -> ColumnCapabilitiesImpl.merge(capabilities, merged.getOrElse(name, null), coercionLogic)
    }.toArray
```
no need for
```scala
result.toArray
```